atom/browser/ui/certificate_trust_mac.mm

```diff
@@ -47,14 +47,14 @@ - (void)dealloc {
   [super dealloc];
 }
 
-- (id)initWithCallback:(const certificate_trust::ShowTrustCallback&)callback
-                 panel:(SFCertificateTrustPanel*)panel
-                  cert:(const scoped_refptr<net::X509Certificate>&)cert
-                 trust:(SecTrustRef)trust
-             certChain:(CFArrayRef)certChain
-             secPolicy:(SecPolicyRef)secPolicy {
+- (id)initWithPromise:(atom::util::Promise)promise
+                panel:(SFCertificateTrustPanel*)panel
+                 cert:(const scoped_refptr<net::X509Certificate>&)cert
+                trust:(SecTrustRef)trust
+            certChain:(CFArrayRef)certChain
+            secPolicy:(SecPolicyRef)secPolicy {
   if ((self = [super init])) {
-    callback_ = callback;
+    promise_.reset(new atom::util::Promise(std::move(promise)));
```

Comment:
There is no need keeping `promise_` in a `std::unique_ptr`. It is only needed when the stored promise may be empty, but in this class since the promise is guaranteed to be always available, it can just be stored as normal value.


```objective-c++
@interface TrustDelegate : NSObject {
 @private
  atom::util::Promise promise_;
}

promise_ = std::move(promise);
```

Reply:
I just realized there is no constructor in objective-c classes so `std::unique_ptr` is the only way to go. This PR should be ready to go!